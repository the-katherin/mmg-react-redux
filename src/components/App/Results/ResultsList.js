import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from 'react-redux';
import { selectTopResults } from '../../../selectors'


export class ResultsList extends Component {

    render() {

        const topResults = this.props.topResults;


        return (
            <ol className='results__list'>
                {
                    topResults.map(
                        (item, index) =>
                            <li key={index}>
                                {item.username}: {item.score}
                            </li>)
                }
            </ol>
        );
    }
};


ResultsList.propTypes = {
    topResults: PropTypes.oneOfType([PropTypes.array, PropTypes.string]).isRequired,
};


export default connect(
    selectTopResults
)(ResultsList);