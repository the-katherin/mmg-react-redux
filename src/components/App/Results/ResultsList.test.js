import React from "react";
import { ResultsList } from "./ResultsList";

let wrapper;


describe("ResultsList", () => {
    beforeEach(() => {
        wrapper = shallow(<ResultsList
            topResults={[]}
        />)
    });

    test("match snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });


});
