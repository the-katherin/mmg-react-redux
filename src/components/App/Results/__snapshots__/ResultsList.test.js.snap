// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList match snapshot 1`] = `
<ol
  className="results__list"
/>
`;
